static/js/containers/ChannelPage.js

```diff
@@ -16,41 +16,68 @@ import { getChannelName } from "../lib/util"
 
 import type { Dispatch } from "redux"
 import type { Match } from "react-router"
+import type { RestState } from "../flow/restTypes"
+import type { Channel, Post } from "../flow/discussionTypes"
 
 class ChannelPage extends React.Component {
   props: {
     match: Match,
     dispatch: Dispatch,
-    channels: Object,
-    postsForChannel: Object,
-    posts: Object
+    channels: RestState<Map<string, Channel>>,
+    postsForChannel: RestState<Map<string, Array<string>>>,
+    posts: RestState<Map<string, Post>>,
+    subscriptions: RestState<Array<Channel>>,
```

Comment:
Any thoughts about renaming `channels` to `channel`/`channelDetail` et al, and `subscriptions` to `subscribedChannels`? I think either or both of those would be a lot clearer, and I think it's confusing enough to be annoying about it in this review 😁